compiler/compiler_test.go

```diff
@@ -232,7 +233,7 @@ func TestCompile(t *testing.T) {
 	}
 
 	for _, test := range tests {
-		program, err := expr.Compile(test.input, expr.Env(Env{}))
+		program, err := expr.Compile(test.input, expr.Env(Env{}), expr.Optimize(false))
```

Comment:
Optimizations needed to be disabled in order for the string, int, and bool comparisons to still be checking the compiler's output given an unoptimized tree.  If optimizations are enabled, they each only have `vm.OpTrue` for the bytecode.